src/lib/updating-element.ts

```diff
@@ -462,7 +483,48 @@ export abstract class UpdatingElement extends HTMLElement {
    * @returns {Element|DocumentFragment} Returns a node into which to render.
    */
   protected createRenderRoot(): Element|ShadowRoot {
-    return this.attachShadow({mode : 'open'});
+    const shadowRoot = this.attachShadow({mode : 'open'});
+    this.createRenderRootStyles(shadowRoot);
+    return shadowRoot;
+  }
+
+  /**
+   * Applies styling to the element shadowRoot using the `static get styles`
+   * property. Styling will apply using `shadowRoot.adoptedStyleSheets` where
+   * available and will fallback otherwise.
+   */
+  protected createRenderRootStyles(shadowRoot: ShadowRoot) {
+    let styles = (this.constructor as typeof UpdatingElement).styles;
+    // de-duplicate styles preserving the last item in the list.
+    const stylesSet = new Set();
```

Comment:
Not sure if this is better, but it gets rid of the deletes (which involve hashes) in exchange for a reverse:

```js
const stylesSet = new Set();
for (const i = styles.length - 1; i >= 0; i==) {
  s.add(styles[i]);
}
styles = [...styleSet].reverse();
```

or:

```js
styles = [...new Set([...styles].reverse())].reverse();
```

Reply:
Used `reduceRight` + `reverse`